### DISCUSSION TITLE
V3: Issue with pagination + wire:navigate

### DISCUSSION THREAD

**Discussion by benjamincrozat** [livewire/livewire#5837] [7. Bugs (archived)] [answered]
Hello,

I'm testing Livewire v3 and putting `wire:navigate` everywhere to see if it's viable.

I also published the views for pagination and added `wire:navigate` in them too, but this is where I stumbled upon an unexpected behavior.

I noticed that the new content is fetched without the query string (`?page=2`), which explains why nothing happens.

Is this something that you want to be fixed? If it is, I can send the PR myself if you quickly tell me where I should look.

**Comment by joshhanley** [comment]
@benjamincrozat yeah I think that is something that needs to be fixed. If you can submit a PR that would be great! Not sure exactly where exactly, but have a look at `features/SupportNavigate`. Thanks!

**Reply by benjamincrozat** [reply]
I managed to fix the issue by adding `+ destination.search` wherever there was `destination.pathname` in _livewire.js_. That being said, I doubt this is the correct way to do it. But I cannot find a non-concatenated version of the code.

**Reply by benjamincrozat** [reply]
(Oh, right, I modified what's in _./vendor/livewire/livewire_… 🤦‍♂️)

**Reply by benjamincrozat** [reply]
Even after cloning the repo, the only place I can do the same modifications is in _dist/livewire.js_. Would love to get more guidance if you don't mind.

**Reply by PhiloNL** [reply]
You should be good to go like this:

In your Laravel project composer.json add the repository path to your livewire fork:

```
    "repositories": [
        {
            "type": "path",
            "url": "../livewire"
        }
    ],
```

Run `composer update` to symlink the directory.
From your fork directory e.g. `livewire` run `npm install && npm run watch`.
Now you can make changes to the javascript and it will compile automatically for you to test in your Laravel project.

**Reply by nuernbergerA** [reply]
@PhiloNL 
>  run `npm install && npm run watch`

as far is i know this will not work currently.
the alpine stuff (alpinejs branch: livewire3) is not yet published on npm and also behind the master

![image](https://github.com/livewire/livewire/assets/13331388/33098d9c-8c78-4dc4-a7c8-a5515e8e7665)

i would recommend to wait for caleb to sort the alpine js stuff

**Reply by PhiloNL** [reply, marked as answer]
If you want to modify the Alpine components, you can do so as well.

```
gh repo clone alpinejs/alpine
git checkout livewire3

npm install && npm run watch
```

Next, `cd` into the plugin you want to modify e.g. `cd packages/navigate/` and run `npm link`.
Now switch to your Livewire fork and run `npm link @alpine/navigate` and run `npm run watch` on the Livewire fork.
You will now be using the local instance of that Alpine package. If you make changes it will compile and the Livewire fork will automatically compile the livewire dist file.

**Reply by benjamincrozat** [reply]
Thanks a lot guys! Makes perfect sense now! I'll go create my PR on https://github.com/alpinejs/alpine/tree/livewire3
